pimd/pim_mroute.c

```diff
@@ -232,6 +232,8 @@ static int pim_mroute_msg_nocache(int fd, struct interface *ifp,
 		pim_upstream_mroute_iif_update(up->channel_oil, __func__);
 	}
 	pim_register_join(up);
+	/* if we have receiver, inherit from parent */
+	pim_upstream_inherited_olist_decide(pim_ifp->pim, up);
```

Comment:
why is this not being properly handled in the upstream creation just above?

Reply:
Presently inheritence and upstream creation is kept seperate.
Ther could be scenario where inheritence not required in upstream. IMHO, we need not introduce that intelligence into upstream creation.